wordpress-mobile/WordPress-Android
# Update site sharing menu visibility

Fixes #14944

This PR updates "My Site" -> "Sharing" menu visibility for Jetpack sites as well as for other conditions taken from the iOS app in the FluxC PR: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2047

### To test

It should be sufficient to see if FluxC PR includes important tests for the "Sharing" menu support.
I've also tested important conditions manually using the below steps (you may randomly pick a scenario and see it works as expected):

PS: Tested using WordPress app build variant for non Jetpack sites

`Jetpack Site`

1. Login with a wpcom account having a site with a jetpack complete plan e.g. [pressable-jetpack-complete](https://pressable-jetpack-complete.mystagingwebsite.com/) on Calypso as well in the app.
2. Go to Jetpack -> Settings -> Sharing on Calypso

    <img width="400" alt="Screenshot 2021-07-02 at 9 37 40 AM" src="https://user-images.githubusercontent.com/1405144/124221209-ed325c80-db1c-11eb-8347-ed5561295bc0.png">
3. Based on the enabled state of "Publicize connections" and "Sharing buttons" on Calypso, check that sharing menu is shown/hidden appropriately on the "My Site" tab in the app.

    Publicize connections | Sharing buttons | Sharing Menu Visibility
    ---------------------- |------------------|-----------------------
    Enabled  | Disabled  | Shown 
    Disabled| Enabled | Shown
    Disabled| Disabled | Hidden
    Enabled| Enabled | Shown

`WpCom site (with publish_posts capabilities)`
- Login with a wpcom account in the app having a site having publish_posts capabilities (admin/ editor/ author user roles)
- Switch to the site using the site picker.
- Notice that My Site -> Sharing menu is shown.

`WpCom Site (no publish_posts, manage_options capabilities)`

- Login with a wpcom account in the app having a site with you added as a contributor (Contributor can write and edit their own posts but can't publish them).
- Switch to the site using the site picker.
- Notice that My Site -> Sharing menu is hidden.

`Self-hosted non Jetpack site`

- Login using a Jurassic ninja self-hosted in the app.
- Notice that My Site -> Sharing menu is hidden.

`Self-hosted Jetpack site`

- Login using a Jurassic ninja self-hosted in the app.
- Install Jetpack and connect to wpcom account.
- Notice that My Site -> Sharing menu is hidden for the site.
- Enable "Sharing buttons" Jetpack module for the site on Tools -> Sharing Buttons in Calypso.
    
    <img width="400" alt="Screenshot 2021-07-02 at 10 18 29 AM" src="https://user-images.githubusercontent.com/1405144/124222800-11dc0380-db20-11eb-8e00-9f753a80dfd5.png">

- Refresh sites using site picker in the app, re-select the site.
- Notice that the My Site -> Sharing menu is now shown for the site.

### Merge Instructions

- Review corresponding FluxC PR 
- Merge the FluxC PR to develop, create a tag, update it in the WPAndroid build.gradle  -  I'll address it after the PR is approved
- Remove Not Ready for Merge label
- Merge this PR

## Regression Notes
1. Potential unintended areas of impact 
Sharing menu visibility for `wpcom` sites.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested "Sharing" menu visibility for different types of sites/ Jetpack active modules/ user roles manually.

3. What automated tests I added (or what prevented me from doing so)
Added tests in the FluxC PR.

PR submission checklist:
